core/src/main/java/org/apache/accumulo/core/classloader/ClassLoaderUtil.java

```diff
@@ -36,25 +39,30 @@ private ClassLoaderUtil() {
   /**
    * Initialize the ContextClassLoaderFactory
    */
-  public static synchronized void initContextFactory(AccumuloConfiguration conf) {
+  public static synchronized void initContextFactory(final AccumuloConfiguration conf) {
     if (FACTORY == null) {
       LOG.debug("Creating {}", ContextClassLoaderFactory.class.getName());
       String factoryName = conf.get(Property.GENERAL_CONTEXT_CLASSLOADER_FACTORY);
       if (factoryName == null || factoryName.isEmpty()) {
-        // load the default implementation
         LOG.info("Using default {}, which is subject to change in a future release",
             ContextClassLoaderFactory.class.getName());
-        FACTORY = new DefaultContextClassLoaderFactory(conf);
-      } else {
-        // load user's selected implementation
-        try {
-          var factoryClass = Class.forName(factoryName).asSubclass(ContextClassLoaderFactory.class);
-          LOG.info("Creating {}: {}", ContextClassLoaderFactory.class.getName(), factoryName);
-          FACTORY = factoryClass.getDeclaredConstructor().newInstance();
-        } catch (ReflectiveOperationException e) {
-          throw new IllegalStateException("Unable to load and initialize class: " + factoryName, e);
-        }
+        factoryName = DefaultContextClassLoaderFactory.class.getName();
+      }
+
+      try {
+        var factoryClass = Class.forName(factoryName).asSubclass(ContextClassLoaderFactory.class);
+        LOG.info("Creating {}: {}", ContextClassLoaderFactory.class.getName(), factoryName);
+        FACTORY = factoryClass.getDeclaredConstructor().newInstance();
+      } catch (ReflectiveOperationException e) {
+        throw new IllegalStateException("Unable to load and initialize class: " + factoryName, e);
       }
+
+      FACTORY.setEnvironment(new ContextClassLoaderEnvironment() {
+        @Override
+        public ServiceEnvironment.Configuration getConfiguration() {
+          return new ConfigurationImpl(conf);
+        }
+      });
```

Comment:
This can be a lambda instead of an anonymous class.

Reply:
got it